### ISSUE TITLE
Terminated salespeople shouldn't be allowed to be added to sales, but should be allowed for CRM, MCR, MDB

### ISSUE THREAD

**Issue by kbower** [kbower/tickettest1#989]
_Imported from trac ticket 989.
Created by **rneville**
Opened in trac: Aug 10, 2011, 9:18:14 AM
Last modified in trac: May 22, 2013, 9:59:03 PM_
***
_by rneville:_
Good morning, you should not allow a salesperson to be added to a sale if the written date of the sale is greater than the end date of EMP$EMP_TP.END_DT where EMP$EMP_TP.EMP_TP_CD = ‘SLS’. However, they can still be added to a CRM, MCR or MDB.

**Comment by kbower** [comment]
_**kbower** trac comment on Aug 10, 2011, 9:19:33 AM:_

Thought we were already checking very similar...

**Comment by kbower** [comment]
_**rneville** trac comment on Jun 14, 2012, 2:32:16 PM:_

Calvin, see one of us for explanation before you start on this one...

**Comment by kbower** [comment]
_**kbower** trac comment on May 16, 2013, 9:04:01 PM:_

This is already prevented for sales and orderdetails:
```python
        Salesperson.check_for_obj(self, 'salesperson1')
        Salesperson.check_for_obj(self, 'salesperson2')
```
What it isn't doing is allowing it for Order.RETURN, Order.CREDIT, Order.DEBIT

**Comment by kbower** [comment]
_**criley** trac comment on May 17, 2013, 1:38:25 PM:_

(Changeset [[4152]](../commit/36e3d946194b0c55bf11bb7a2c29dbcd13216ea6))
* Fixes #989 (Terminated salespeople shouldn't be allowed to be added to sales, but should be allowed for CRM, MCR, MDB) #all

**Comment by kbower** [comment]
_**criley** trac status change on May 17, 2013, 1:38:25 PM:_ **closed**

**Comment by kbower** [comment]
_**kbower** trac comment on May 17, 2013, 1:47:55 PM:_

We still need the check for orderdetails and _all_ order types to make sure the saleperson is _valid_, just that for orders of `isinstance(self, _AdjustingOrder)`, we skip the is_active check.

```python
def check_for_obj(cls, obj, field):
```
Will need to become:
```python
def check_for_obj(cls, obj, field, check_active):
```
Passing a boolean `check_active`

**Comment by kbower** [comment]
_**kbower** trac status change on May 17, 2013, 1:47:55 PM:_ **reopened**

**Comment by kbower** [comment]
_**kbower** trac comment on May 17, 2013, 1:49:41 PM:_

(Changeset [[4153]](../commit/3cf5e9ac418c0efddded9babc5c71e7a8045c3aa))
* revert [r4152](../commit/36e3d946194b0c55bf11bb7a2c29dbcd13216ea6) to keep code stable, re #989

**Comment by kbower** [comment]
_**kbower** trac comment on May 22, 2013, 9:25:45 PM:_

see [r4164](../commit/60da2dd14ec9f4e307fc31e01e4798968b27001b)

**Comment by kbower** [comment]
_**kbower** trac comment on May 22, 2013, 9:59:03 PM:_

(Changeset [[4166]](../commit/b6615d0d0dfaaa5b28f0c52084b6ea44e09f8771))
* closes #989, Terminated salespeople shouldn't be allowed to be added to sales, but should be allowed for CRM, MCR, MDB #all
   * Note that we do allow salespeople to be added as long as the termination date is prior to orderdate, even if not enabled
   * refactor to skip check for all _AdjustingOrder types and their orderdetails

**Comment by kbower** [comment]
_**kbower** trac status change on May 22, 2013, 9:59:03 PM:_ **closed**
